service_configuration_lib/spark_config.py

```diff
@@ -250,6 +250,35 @@ def _append_sql_shuffle_partitions_conf(spark_opts: Dict[str, str]) -> Dict[str,
     return spark_opts
 
 
+def _get_dra_configs(spark_opts: Dict[str, str]) -> Dict[str, str]:
+    if ('spark.dynamicAllocation.enabled' not in spark_opts or
+        str(spark_opts['spark.dynamicAllocation.enabled']) != 'true'):
+        return spark_opts
+
+    spark_opts['spark.dynamicAllocation.shuffleTracking.enabled'] = "true"
+    spark_opts['spark.dynamicAllocation.executorAllocationRatio'] = "0.8"
+
+    if 'spark.dynamicAllocation.minExecutors' not in spark_opts:
+
+        min_executors = int(spark_opts.get('spark.executor.instances', DEFAULT_EXECUTOR_INSTANCES)) // 4
+        if 'spark.dynamicAllocation.initialExecutors' in spark_opts:
+            min_executors = min(min_executors, int(spark_opts.get('spark.dynamicAllocation.initialExecutors')))
+
+        spark_opts['spark.dynamicAllocation.minExecutors'] = str(min_executors)
+
+    if 'spark.dynamicAllocation.maxExecutors' not in spark_opts:
+
+        max_executors = int(spark_opts.get('spark.executor.instances', DEFAULT_EXECUTOR_INSTANCES))
+        if 'spark.dynamicAllocation.initialExecutors' in spark_opts:
+            min_executors = max(max_executors, int(spark_opts.get('spark.dynamicAllocation.initialExecutors')))
+
+        spark_opts['spark.dynamicAllocation.maxExecutors'] = str(max_executors)
```

Comment:
What was the conclusion on the caching dra config? Wouldn't we loose the benefit of DRA if we keep the executors alive because of the present cache?
Maybe add a high time value and/or add a suggestion?

Reply:
Sounds reasonable. 
* Add as a global variable
* Make sure it could be overridden if given as a config